lib/changelogerator.rb

```diff
@@ -74,46 +78,72 @@ def self.changes_files_in_paths?(change, paths)
   # Optional named parameters:
   # token: a Github personal access token
   # prefix: whether or not to prefix PR numbers with their repo in the changelog
-  def initialize(github_repo, from, to, token: '', prefix: nil)
+  def initialize(github_repo, from, to, token: "", prefix: nil)
     @repo = github_repo
-    @priorities = self.class.priorities
     @gh = Octokit::Client.new(
-      access_token: token
+      access_token: token,
     )
+    @repository = @gh.repository(@repo)
     @prefix = prefix
     @changes = prs_from_ids(pr_ids_from_git_diff(from, to))
-    # add priority to each change
-    @changes.map { |c| apply_priority_to_change(c) }
-  end
+    @changes.map do |c|
+      compute_change_meta(c)
+    end
 
-  def changes_with_label(label)
-    self.class.changes_with_label(@changes, label)
-  end
-
-  def runtime_changes?
-    nil
+    compute_global_meta()
   end
 
   def add(change)
-    changes.prepend(prettify_title(apply_priority_to_change(change)))
+    compute_change_meta(change)
+    prettify_title(change)
+    changes.prepend(change)
+    @meta = compute_global_meta()
   end
 
+  # Add a pull request from id
   def add_from_id(id)
     pull = @gh.pull_request(@repo, id)
     add pull
   end
 
+  def to_json
+    opts = {
+      array_nl: "\n",
+      object_nl: "\n",
+      indent: "  ",
+      space_before: " ",
+      space: " ",
+    }
+    obj = @changes
+
+    commits = {
+        meta: @meta,
+        repository: @repository.to_h,
+        changes: obj.map(&:to_h),
+    }
+
+    JSON.fast_generate(commits, opts)
```

Comment:
It's not clear to me from the rubydocs what the functional difference is between `generate` and `fast_generate`. What motivated your usage of it here?

Reply:
I tested the fast version and it worked :)
I think the fast version does not serialize that deep.